src/reducers/notifications/types.ts

```diff
@@ -0,0 +1,93 @@
+import { AppError } from '../../lib/errors'
+import { NotificationHandler } from '../../ui/notifications/types'
+
+export enum NotificationType {
+  error = 'error',
+  info = 'info',
+  warning = 'warning',
+}
+
+/**
+ * Depending on the notification severity, it might be rendered differently on the UI. For example:
+ * low - simple top notification banner
+ * medium - top notification banner with accent colors on title and action button
+ * high - full screen overlay with call to action
+ */
+
+export enum NotificationSeverity {
+  low = 'low',
+  medium = 'medium',
+  high = 'high',
+}
+
+/**
+ * This allows for some notifications to be handled locally, in a component, in some unique way
+ * local - the notification won't be picked up by the main application notification handler (which would have
+ * resulted in a top banner notification), and can instead be handled by another component
+ * global - the notification will be processed by the main application handler. A top banner will render,
+ * after which the notification will be removed from the queue
+ */
+
+export enum NotificationScope {
+  local = 'local',
+  global = 'global',
+}
```

Comment:
Not sure if I'm getting the purpose of that scope, can you give an example. I'd be afraid that global and local notification could possible overlap each other...

Reply:
Notifications and errors can be handled globally (i.e. by the top bar notification handler), or locally (i.e. the orange border on the camera screen in case of failed scans). I was considering settings things up so all notifications of `low` severity are handled locally, but that limits the notifications we can represent. Given our current setup, a top bar notification with normal colors is currently - `scope: global, severity: low`
a top bar notification with accent colors can be handled by `scope: global, severity: medium` 
and a full screen warning / info / notification would have `scope: global, severity: high`.

Without the `scope` key, we can't easily handle notifications handled outside of the global handler.

In the QR code scanner case, local notifications of various severity can be handled differently, i.e. with different border colors / button activity, etc.

Reply:
I believe this is obsolete/resolved after the discussion we had: https://github.com/jolocom/smartwallet-app/issues/1499#issuecomment-552515594